api/v1alpha4/openstackcluster_webhook.go

```diff
@@ -77,6 +77,18 @@ func (r *OpenStackCluster) ValidateUpdate(oldRaw runtime.Object) error {
 		allErrs = append(allErrs, field.Forbidden(field.NewPath("spec", "identityRef", "kind"), "must be a Secret"))
 	}
 
+	// Allow changes to Spec.IdentityRef.Name.
+	if old.Spec.IdentityRef != nil && r.Spec.IdentityRef != nil {
+		old.Spec.IdentityRef.Name = ""
+		r.Spec.IdentityRef.Name = ""
+	}
```

Comment:
I *think* this is ok. Changing IdentityRef.Name is definitely a potential foot gun, but:
* I can't think of any reason to strictly forbid it
* There may be edge cases when it's useful to support allowing changing it

I'd appreciate a second opinion from anybody familiar with the original design decisions!

Reply:
I was not part of the original design decisions but I also think it should be ok to allow changes.

E.g. maybe somebody wants to keep the old credential information and creating a new secret for the new credential.

Reply:
CAPA is only disallowing the removal of a IdentityRef

https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/api/v1beta1/awscluster_webhook.go#L136-L142

Reply:
IIRC the identityref is only used during reconcile to get the OpenStack creds and then access the OpenStack API with them. (aka it is not used to store them somewhere in the cluster).

If I'm correct, I see no reason to deny changes to the identityRef

Reply:
That's also my understanding.